### ISSUE TITLE
js file still generated when not in "files" and ignored by "filesGlob" (and "compileOnSave" is off)

### ISSUE THREAD

**Issue by justin-calleja** [TypeStrong/atom-typescript#984]
I'm new to Typescript so this might not be the best setup.

I have 2 separate build steps, one for tests, one for the actual thing.

My scripts:

```json
  "scripts": {
    "clean": "rm -rf ./lib && mkdir ./lib",
    "cleanTest": "rm -rf ./testlib && mkdir ./testlib",
    "build": "npm run clean && tsc -p ./tsconfig.json",
    "buildTest": "npm run cleanTest && tsc -p ./tsconfig-test.json",
    "prepublish": "npm run test",
    "pretest": "npm run build && npm run buildTest",
    "test": "mocha --ui tdd ./test/index.js"
  },
```

i.e. I do `tsc -p ./tsconfig.json` and `tsc -p ./tsconfig-test.json` depending on what I want to build.

I'm trying to start writing test/index.ts but every time I save, I get test/index.js generated. This is not what I want to do. I want my test js files in `testlib` and main js files in `lib` (so I can clean them off safely with `rm -rf <dir>`).

I don't expect test/index.ts to be compiled because:

1. tsconfig.json file globs `src/**/*.ts`, paying no attention to `test` dir (and `"files"` reflects this).
2. I tried setting `compileOnSave` to false but it's still generating `test/index.js`.

I imagine having 2 different compilation set-ups will make code intellicense not work for one or the other (e.g. not work for test *.ts files).

Is this the right way of going about this problem?

Does the solution depend on #492 ?

**Comment by basarat** [comment]
> ignored by "filesGlob"

Almost certainly not. Might be a bug in our code where it should be ignored but wasn't. But most likely included due to compilation context expansion.

**Comment by basarat** [comment]
please provide the tsconfig :rose:

**Comment by justin-calleja** [comment]
Hi basarat.

I had deleted this approach (2 separate compilation contexts: one for src/ and one fro test/) and had decided to stick to JS for writing tests.

I'm not sure how I was getting test/index.js generated every time I saved test/index.ts.

I have created a dummy repo with what I think I had tried at the time here: https://github.com/justin-calleja/atom-typescript-984

I'm closing as I cannot reproduce the issue.
